From e096941b5c3a261f33fe443b7cad8260cbf1bee1 Mon Sep 17 00:00:00 2001
From: Larisa Grigore <larisa.grigore@nxp.com>
Date: Tue, 11 May 2021 17:30:36 +0300
Subject: [PATCH 231/269] s32g: Disable PMIC before clocks init

In this was we can disable PMIC ~27ms earlier.

Issue: ALB-6995
Upstream-Status: Pending 

Signed-off-by: Larisa Grigore <larisa.grigore@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/nxp/s32g/s32g_bl31.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/plat/nxp/s32g/s32g_bl31.c b/plat/nxp/s32g/s32g_bl31.c
index 4226a3b12..ab4cd31c0 100644
--- a/plat/nxp/s32g/s32g_bl31.c
+++ b/plat/nxp/s32g/s32g_bl31.c
@@ -402,12 +402,6 @@ void bl31_platform_setup(void)
 {
 	int ret;
 
-	update_core_state(plat_my_core_pos(), 1);
-	s32g_gic_setup();
-
-	dt_clk_init();
-	s32g_correct_qspi_clock();
-
 	generic_delay_timer_init();
 
 	dt_init_pmic();
@@ -416,6 +410,12 @@ void bl31_platform_setup(void)
 	ret = pmic_disable_wdg();
 	if (ret)
 		ERROR("Failed to disable VR5510 watchdog\n");
+
+	update_core_state(plat_my_core_pos(), 1);
+	s32g_gic_setup();
+
+	dt_clk_init();
+	s32g_correct_qspi_clock();
 }
 
 /* TODO: Last-minute modifications before exiting BL31:
-- 
2.17.1

